process/geometry/tfcoil_geometry.py

```diff
@@ -0,0 +1,191 @@
+    tfcth: float,
+) -> List[RectangleGeometry]:
+    """Calculates rectangular geometries for tf coils in a picture frame/rectangular shape parametrization
+
+    :param x1: radial location of arc point 1
+    :type x1: float
+    :param x2: radial location of arc point 2
+    :type x2: float
+    :param x4: radial location of arc point 4
+    :type x4: float
+    :param x5: radial location of arc point 5
+    :type x5: float
+    :param y1: vertical location of arc point 1
+    :type y1: float
+    :param y2: vertical location of arc point 2
+    :type y2: float
+    :param y4: vertical location of arc point 4
+    :type y4: float
+    :param y5: vertical location of arc point 5
+    :type y5: float
+    :param tfcth: inboard tf coil thickness
+    :type tfcth: float
+    :return: list of RectangleGeometry - dataclass returning rectangular geometry parameters
+    :rtype: List[RectangleGeometry]
+    """
+    return_rects = []
+    return_rects.append(
+        RectangleGeometry(
+            center_x=x5 - tfcth,
+            center_z=y5 - tfcth,
+            width=tfcth,
+            height=(y1 - y5 + 2.0 * tfcth),
+        )
+    )
+    return_rects.append(
+        RectangleGeometry(
+            center_x=x4,
+            center_z=y4 - tfcth,
+            width=tfcth,
+            height=(y2 - y4 + 2.0 * tfcth),
+        )
+    )
+    return_rects.append(
+        RectangleGeometry(center_x=x5, center_z=y5 - tfcth, width=x4 - x5, height=tfcth)
+    )
+    return_rects.append(
+        RectangleGeometry(center_x=x1, center_z=y1, width=(x2 - x1), height=tfcth)
+    )
+
+    return return_rects
+
+
+def tfcoil_geometry_d_shape(
+    x1: float,
+    x2: float,
+    x3: float,
+    x4: float,
+    x5: float,
+    y1: float,
+    y2: float,
+    y4: float,
+    y5: float,
+    tfcth: float,
+    rtangle: float,
+    rtangle2: float,
+) -> Tuple[List[RectangleGeometry], list]:
+    """Calculates radial and vertical distances for the geometry of the tf coils in a D-shape parametrization
+
+    :param x1: radial location of arc point 1
+    :type x1: float
+    :param x2: radial location of arc point 2
+    :type x2: float
+    :param x3: radial location of arc point 3
+    :type x3: float
+    :param x4: radial location of arc point 4
+    :type x4: float
+    :param x5: radial location of arc point 5
+    :type x5: float
+    :param y1: vertical location of arc point 1
+    :type y1: float
+    :param y2: vertical location of arc point 2
+    :type y2: float
+    :param y4: vertical location of arc point 4
+    :type y4: float
+    :param y5: vertical location of arc point 5
+    :type y5: float
+    :param tfcth: inboard tf coil thickness
+    :type tfcth: float
+    :param rtangle: angle used in tf coil parametrization
+    :type rtangle: float
+    :param rtangle2: angle used in tf coil parametrization
+    :type rtangle2: float
+    :return: radial and vertical coordinates for tf coils
+    :rtype: Tuple[List[RectangleGeometry], list]
```

Comment:
`:rtype: Tuple[List[RectangleGeometry], List[float]]`? Double check that but I think thats a more descriptive type

Reply:
Wait no maybe its a List[List[float]] looking at the code. Need a better description of the types and meanings of the variables involved